File: java/src/main/java/com/workshop/architecture/fitness/presentation/ActivateMembershipResponse.java
package com.workshop.architecture.fitness.presentation;

import java.time.LocalDate;

public record ActivateMembershipResponse(
        String membershipId,
        String customerId,
        String planId,
        int planPrice,
        int planDuration,
        String status,
        LocalDate startDate,
        LocalDate endDate,
        String invoiceId,
        String externalInvoiceId,
        LocalDate invoiceDueDate
) {
}

File: python/src/workshop_api/fitness/clean_architecture/adapter/controller/activate_membership_response.py
from datetime import date

from pydantic import BaseModel, ConfigDict, Field


class ActivateMembershipResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    membership_id: str = Field(alias="membershipId", serialization_alias="membershipId")
    customer_id: str = Field(alias="customerId", serialization_alias="customerId")
    plan_id: str = Field(alias="planId", serialization_alias="planId")
    plan_price: int = Field(alias="planPrice", serialization_alias="planPrice")
    plan_duration: int = Field(alias="planDuration", serialization_alias="planDuration")
    status: str
    start_date: date = Field(alias="startDate", serialization_alias="startDate")
    end_date: date = Field(alias="endDate", serialization_alias="endDate")
    invoice_id: str = Field(alias="invoiceId", serialization_alias="invoiceId")
    external_invoice_id: str = Field(
        alias="externalInvoiceId",
        serialization_alias="externalInvoiceId",
    )
    invoice_due_date: date = Field(alias="invoiceDueDate", serialization_alias="invoiceDueDate")
